spring-core/src/main/java/org/springframework/core/annotation/AnnotationsScanner.java

```diff
@@ -234,9 +234,7 @@ private AnnotationsScanner() {
 		if (includeEnclosing && enclosingClass != null) {
 			R enclosingResult = processClassHierarchy(context, aggregateIndex,
 					enclosingClass, processor, classFilter, includeInterfaces, true);
-			if (enclosingResult != null) {
-				return enclosingResult;
-			}
+			return enclosingResult;
```

Comment:
Although this may be technically equivalent, it breaks with the structure of the rest of the method.

As such, I do not think the change is warranted.

So, please revert this change.